internal/usage/usage.go

```diff
@@ -0,0 +1,96 @@
+package usage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/grafana/synthetic-monitoring-agent/internal/feature"
+	sm "github.com/grafana/synthetic-monitoring-agent/pkg/pb/synthetic_monitoring"
+)
+
+// Report represents a specific usage event that will be sent to https://stats.grafana.com to be processed and stored.
+// Each attribute represents a column in a BigQuery table that can be easily searched.
+// Adding new attributes to Report will not automatically update the table, and instead needs to be handled in https://github.com/grafana/usage-stats
+type Report struct {
+	CreatedAt    string `json:"createdAt"`
+	OS           string `json:"os"`
+	Arch         string `json:"arch"`
+	Report       string `json:"report"`
+	Version      string `json:"version"`
+	Public       bool   `json:"public"`
+	Features     string `json:"features"`
+	UsageStatsId string `json:"usageStatsId"`
+	TenantID     int64  `json:"tenantId"`
+}
+
+// UsageReporter represents
+type UsageReporter struct {
```

Comment:
Since the package name is `usage`, should this be named `Reporter` so we avoid the repetition on `usage.UsageReporter`?